app/controllers/flag.controller.js

```diff
@@ -0,0 +1,35 @@
+var authentication = require('../utils/authentication.utils');
+var Flag = require('mongoose').model('Flag');
+var Ticket = require('mongoose').model('Ticket');
+var User = require('mongoose').model('User');
+
+exports.flagListing = function(req, res, next) {
+    var listing = (req.body.listing) ? (req.body.listing) : null;
```

Comment:
You do not need to do this ternary operator here. You are essentially checking if it is null false or undefined, then setting it to null if it is. Just do this:

`var listing = req.body.listing;`